aci/volumes.go

```diff
@@ -164,7 +165,47 @@ func errorEvent(resource string) progress.Event {
 	}
 }
 
+func checkVolumeUsage(ctx context.Context, aciContext store.AciContext, id string) error {
+	resourceGroups, err := NewACIResourceGroupHelper().ListGroups(ctx, aciContext.SubscriptionID)
+	if err != nil {
+		return err
+	}
+	for _, rg := range resourceGroups {
+		if rg.Name == nil {
+			continue
+		}
+		containerGroups, err := getACIContainerGroups(ctx, aciContext.SubscriptionID, *rg.Name)
+		if err != nil {
+			return err
+		}
+		for _, cg := range containerGroups {
+			if hasVolume(cg.Volumes, id) {
+				return errors.Errorf("Volume %q is used in container group %q inside resource group %q",
+					id, *cg.Name, *rg.Name)
+			}
+		}
+	}
+	return nil
+}
+
+func hasVolume(volumes *[]containerinstance.Volume, id string) bool {
+	if volumes == nil {
+		return false
+	}
+	for _, v := range *volumes {
+		if v.AzureFile != nil && v.AzureFile.StorageAccountName != nil && v.AzureFile.ShareName != nil &&
+			(*v.AzureFile.StorageAccountName+"/"+*v.AzureFile.ShareName) == id {
+			return true
+		}
+	}
+	return false
+}
+
 func (cs *aciVolumeService) Delete(ctx context.Context, id string, options interface{}) error {
+	err := checkVolumeUsage(ctx, cs.aciContext, id)
```

Comment:
Maybe we should add a `--force` option to avoid the check...

Reply:
maybe... not a blocker for this PR, we can add this later on if we want. cc @nebuk89